src/openssl.cr

```diff
@@ -1,5 +1,56 @@
 require "./openssl/lib_ssl"
 
+# ### Example
```

Comment:
Maybe have a header? Something as basic as `# OpenSSL integration`

Maybe also have an introduction that says that:
- TLS sockets need a context, potentially with keys (required for servers) and configuration.
- TLS sockets will wrap the underlying TCP socket, and any further communication must happen through the `OpenSSL::SSL::Socket` only.

Then continue with:

- the default configuration (sane defaults, uses Mozilla Intermediate, best effort, please contribute)
- the server example —I'm not sure we should add an example about how to generate the certificates, one should buy a certificate instead;
- the client example.